bundles/admin/admin-layereditor/view/AdminLayerForm/AdditionalTabPane.jsx

```diff
@@ -1,37 +1,57 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import PropTypes from 'prop-types';
 import { TextInput, TextAreaInput, Message } from 'oskari-ui';
 import { StyledTab, StyledComponent } from './StyledFormComponents';
 import { LocaleConsumer, Controller } from 'oskari-ui/util';
 
-export const AdditionalTabPane = LocaleConsumer(({ layer, controller, getMessage }) => (
-    <StyledTab>
-        <Message messageKey='metainfoId'/>
-        <StyledComponent>
-            <TextInput
-                placeholder={getMessage('metaInfoIdDesc')}
-                value={layer.metadataid}
-                onChange={(evt) => controller.setMetadataIdentifier(evt.target.value)} />
-        </StyledComponent>
-        <Message messageKey='gfiContent'/>
-        <StyledComponent>
-            <TextAreaInput
-                rows={6}
-                value={layer.gfiContent}
-                onChange={(evt) => controller.setGfiContent(evt.target.value)} />
-        </StyledComponent>
-        <Message messageKey='attributes'/>
-        <StyledComponent>
-            <TextAreaInput
-                rows={6}
-                value={JSON.stringify(layer.attributes || {}, null, 2)}
-                onChange={(evt) => controller.setAttributes(evt.target.value)} />
-        </StyledComponent>
-    </StyledTab>
-));
+const LayerComposingModel = Oskari.clazz.get('Oskari.mapframework.domain.LayerComposingModel');
+
+export const AdditionalTabPane = LocaleConsumer(({ layer, propertyFields, controller, getMessage }) => {
+    const metainfoInput =
+        <Fragment>
+            <Message messageKey='metainfoId'/>
+            <StyledComponent>
+                <TextInput
+                    placeholder={getMessage('metaInfoIdDesc')}
+                    value={layer.metadataid}
+                    onChange={(evt) => controller.setMetadataIdentifier(evt.target.value)} />
+            </StyledComponent>
+        </Fragment>;
+
+    const gfiContentInput =
+        <Fragment>
+            <Message messageKey='gfiContent'/>
+            <StyledComponent>
+                <TextAreaInput
+                    rows={6}
+                    value={layer.gfiContent}
+                    onChange={(evt) => controller.setGfiContent(evt.target.value)} />
+            </StyledComponent>
+        </Fragment>;
+
+    const attributesInput =
+        <Fragment>
+            <Message messageKey='attributes'/>
+            <StyledComponent>
+                <TextAreaInput
+                    rows={6}
+                    value={JSON.stringify(layer.attributes || {}, null, 2)}
+                    onChange={(evt) => controller.setAttributes(evt.target.value)} />
+            </StyledComponent>
+        </Fragment>;
```

Comment:
Maybe premature optimization but if we would replace metainfoInput, gfiContentInput and attributesInput with method calls or components outside the render function they would not be created even in memory if they are not needed?